src/ejabberd_sm.erl

```diff
@@ -928,7 +907,7 @@ check_for_sessions_to_replace(HostType, JID) ->
     %% replacement for max_sessions. We need to check this at some point.
     ReplacedRedundantSessions = check_existing_resources(HostType, LUser, LServer, LResource),
     AllReplacedSessionPids = check_max_sessions(HostType, LUser, LServer, ReplacedRedundantSessions),
-    [Pid ! replaced || Pid <- AllReplacedSessionPids],
+    [mongoose_c2s:exit(Pid, <<"Replaced by new connection">>) || Pid <- AllReplacedSessionPids],
```

Comment:
After this, we could remove the `handle_info` clause in `mongoose_c2s` that was catching the `replaced` message.